frontend/src/pages/NewRunSwitcher.tsx

```diff
@@ -14,10 +14,31 @@ function NewRunSwitcher(props: PageProps) {
   const namespace = React.useContext(NamespaceContext);
 
   const urlParser = new URLParser(props);
+  const originalRunId = urlParser.get(QUERY_PARAMS.cloneFromRun);
   const pipelineId = urlParser.get(QUERY_PARAMS.pipelineId);
   const pipelineVersionIdParam = urlParser.get(QUERY_PARAMS.pipelineVersionId);
 
-  const { isSuccess, isFetching, data: templateString } = useQuery<string, Error>(
+  const {
+    isSuccess: runIsSuccess,
+    isFetching: runIsFetching,
+    data: templateStrFromRunId,
+  } = useQuery(
+    [originalRunId],
```

Comment:
Do you want to add a constant value `apiRun` or similar to uniquely identify the query cache key?